### PR TITLE
Add TCP check

### PR DESCRIPTION
**Pull request by nfranchet**
TCP Check to be used by sauna

### CODE REVIEW

File: sauna/plugins/ext/tcp.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import socket
+
+from sauna.plugins import (Plugin, bytes_to_human, human_to_bytes,
```
Comment (NicolasLM) [inline sauna/plugins/ext/tcp.py]: Some imports are useless here. You only use Plugin and PluginRegister.

File: sauna/plugins/ext/tcp.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+class Tcp(Plugin):
+
+    def __init__(self, config):
+        super().__init__(config)
+        self.config = {
+            'host': config.get('host', 'localhost'),
+            'port': config.get('port', 11211),
+            'timeout': config.get('timeout', 5)
+        }
+        self._stats = None
```
Comment (NicolasLM) [inline sauna/plugins/ext/tcp.py]: Not used

File: sauna/plugins/ext/tcp.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    @my_plugin.check()
+    def request(self, check_config):
+        try:
+            with socket.create_connection((check_config['host'],
+                                           check_config['port']),
+                                          timeout=check_config['timeout']) as s:
+                if s is None:
```
Comment (NicolasLM) [inline sauna/plugins/ext/tcp.py]: AFAIK `with` construct guaranties that `s` will be a valid socket.

```
try:
    with socket.create_connection(....) as s: 
        pass
except Exception as e:
     return Plugin.STATUS_CRIT, "{}".format(e)
else:
    return Plugin.STATUS_OK, "OK"
```

File: sauna/plugins/ext/tcp.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                return Plugin.STATUS_OK, "OK"
+        except Exception as e:
+            return Plugin.STATUS_CRIT, "%s" % e
+        return Plugin.STATUS_CRIT, "Can't connect"
+
+
+    @staticmethod
+    def config_sample():
+        return '''
+        # Memcached
```
Comment (NicolasLM) [inline sauna/plugins/ext/tcp.py]: Copy and paste ;)

File: sauna/plugins/ext/tcp.py (hunk 3 above)
Comment (nfranchet) [inline sauna/plugins/ext/tcp.py]: Is it better to set return in else than in the with ? I think they are equals, but why to use else ?

Comment (NicolasLM) [inline sauna/plugins/ext/tcp.py]: yes, even better

**Pull request merged**